README.md

```diff
@@ -6,3 +6,188 @@ Distributed service to interact with development boards.
+undergoing active development, we'll build the latest from git.
+
+1. Install required tools/libraries to build boardswarm on a suitable system.
+   As we're pulling the latest from git, we'll need that installed too.
+   Assuming a Debian based system:
+   ```
+   $ sudo apt install rustc git protobuf-compiler libudev-dev
+   ```
+1. Clone the boardswarm repository:
+   ```
+   $ git clone https://github.com/boardswarm/boardswarm.git
+   ```
+1. We can now build boardswarm:
+   ```
+   $ cd boardswarm
+   $ cargo build
+   ```
+
+### Server Install
+
+The "server" is the machine that will have the development boards attached to
+it.
+
+1. Copy the `boardswarm` binary to `/usr/local/bin/` on the server.
+
+1. Create a boardswarm user and group for the boardswarm server to run as:
+   ```
+   $ sudo addgroup --system boardswarm
+   $ sudo adduser --system --ingroup boardswarm --no-create-home --disabled-password boardswarm
+   ```
+
+1. Copy the [example systemd unit](boardswarm/share/boardswarm.service) to
+   `/etc/systemd/system/boardswarm.service`.
+
+1. Copy [example udev rules](boardswarm/share/99-boardswarm.rules) to
+   `/etc/udev/rules.d/`.
+
+1. Reload udev rules and ensure they are applied:
+   ```
+   $ sudo udevadm control --reload-rules
+   $ sudo udevadm trigger
+   ```
+
+1. Copy example [`server.conf`](boardswarm/share/server.conf) to
+   `/etc/boardswarm/server.conf` on target server machine and modify as
+   required.
+
+### Server Configuration
+
+1. Install [pdudaemon](https://github.com/pdudaemon/pdudaemon) and configure it
+   to switch the development boards on and off, using whichever mechanism you
+   are using (ensure that the `server.conf` `pdu` provider is configured to match
+   the configuration set in the pdudaemon configuration):
+   ```
+   $ sudo apt install pdudaemon
+   ```
+
+1. We are going to use Static JWKS based authentication, as documented in the
+   [server README](boardswarm/README.md#static-jwks-based-authentication).
+   Install `rnbyc` so we can create the required tokens:
+   ```
+   $ sudo apt install rnbyc
+   ```
+
+1. Generate an EdDSA key:
+   ```
+   $ rnbyc -j -g  Ed25519  -x   -o private.jwks -p auth.jwks
+   ```
+   Ensure you keep the private key private and safe, it will be needed during
+   client configuration.
+
+1. Copy generated `auth.jwks` to `/etc/boardswarm/`
+   ```
+   $ sudo cp auth.jwks /etc/boardswarm/
+   ```
+
+For each development board you will need a section under `devices`. There is an
+example one provided in the example server configuration which can be used as a
+template.
+
+#### USB Serial Consoles
+
+1. Determine the USB serial port used as the console on the development board.
+
+1. Use the following commmand to determine the required `udev.ID_SERIAL` and
+   `udev.ID_USB_INTERFACE_NUM`:
+   ```
+   $ udevadm info <tty_device> | grep -i 'ID_SERIAL=\|ID_USB_INTERFACE_NUM='
+   ```
+
+#### DFU Booting
```

Comment:
Might make sense  to create a document per protocol like this; as it makes sense to discuss each of them in details (we current support dfu and rockusb volumes)

Reply:
I have worked the device configuration bits into the descriptions already in the server README.md as on reflection that seems a more suitable place to have this, given that there's already a good description of each of these there.

I haven't used rockusb yet...